Dockerfile

```diff
@@ -62,6 +65,7 @@ ENV PYTHON_BASE_IMAGE=${PYTHON_BASE_IMAGE} \
 
 # Install curl and gnupg2 - needed for many other installation steps
 RUN apt-get update \
+    && apt-get install --no-install-recommends -yqq apt-utils >/dev/null 2>&1 \
```

Comment:
Why are we installing this as a separate command and not together with the next step?

Reply:
Because we need curl/gpg2 to run DEV_APT_COMMAND  - and we can potentially override this commmand (and we need curl/gpg2 to be able to install "extra resources" if needed. I prefer to install curl/gpg2 using "official sources" first - to be able to add "new sources" later. Otherwise it would require anyhow to run the "apt update" two times - once with the oriignal sources to update curl/gpg2, downloading keys for the "secondary" sources, and running apt-update again. The optimisations we can get this way are minimal, and it's much better to separate the concerns - the ommand to run all of that in one RUN would be pretty long..

Reply:
However - I think that is a good idea eventually and I merged this RUN with the next one.

Also. I realized that you asked for the

```
    && apt-get install --no-install-recommends -yqq apt-utils >/dev/null 2>&1 \
```

The main reson is that you need to install apt-utils to get rid of some warnings printed by apt-get. But I do not want to redirect stderr to /dev/null for all installation steps (just in case of some errors) - so i only redirect stderr in this one step (otherwise installing apt-utils would produce an stderr warning too).